### PR TITLE
Add `README` to DNS example

### PR DESCRIPTION
**Pull request by Irene-123**
Hello there, I'm a blockchain enthusiast and contributing to open-source projects lately. I've been using substrate lately for my college projects and found that this great DNS smart contract doesn't have a README, so I've added it. Thank you!

### CODE REVIEW

File: examples/dns/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+> This is a port from the blog post's ink! 1.0 based version of the contract to ink! 2.0.
+
+Domain name service contract inspired by
+    [this blog post on medium](https://medium.com/@chainx_org/secure-and-decentralized-polkadot-domain-name-system-e06c35c2a48d).
+
+### Description
+The main function of this contract is domain name resolution which refers to the retrieval of numeric values corresponding to readable and easily memorable names such as "polka.dot" which can be used to facilitate transfers, voting and DApp-related operations instead of resorting to long IP addresses that are hard to remember.
+<br>
+
+![Image](dns_diagram.png)
```
Comment (HCastano) [inline examples/dns/README.md:16]: Thanks for the diagram. Can you make a `examples/dns/.images` folder and put it in there instead?

File: examples/dns/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+   
+
+> This is a port from the blog post's ink! 1.0 based version of the contract to ink! 2.0.
+
+Domain name service contract inspired by
+    [this blog post on medium](https://medium.com/@chainx_org/secure-and-decentralized-polkadot-domain-name-system-e06c35c2a48d).
+
+### Description
+The main function of this contract is domain name resolution which refers to the retrieval of numeric values corresponding to readable and easily memorable names such as "polka.dot" which can be used to facilitate transfers, voting and DApp-related operations instead of resorting to long IP addresses that are hard to remember.
```
Comment (HCastano) [inline examples/dns/README.md:13]: Can you update this to maybe show an example of how this translation would work?

File: examples/dns/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+### Other functionalities to be implemented: 
+- Transfer bidding
+
+> __Note:__<br/>
+   > Depending on your Substrate version you might encounter [a ink_lang dependency error](https://github.com/paritytech/ink/issues/506) which is yet to address. 
+  
+
+
+
+
```
Comment (HCastano) [inline examples/dns/README.md:33]:
```suggestion
```

**Review by HCastano — CHANGES_REQUESTED** [review]
Thanks for you PR, but it needs some fixes.

Also, can you wrap everything to ~90 characters?

File: examples/dns/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- Registration of a new Domain Name
+- Transfer the (already exsiting) owner to new Address
+- Change the Domain Name
+
+### Other functionalities to be implemented: 
+- Transfer bidding
+
+> __Note:__<br/>
+   > Depending on your Substrate version you might encounter [a ink_lang dependency error](https://github.com/paritytech/ink/issues/506) which is yet to address.
```
[the remark this answers is not in the document]
Comment (Irene-123) [reply examples/dns/README.md:28]: I came across this when compiling the contract. While compiling it gives an error of 'ink_lang' 
image

But as mentioned [above](https://github.com/paritytech/ink/pull/1542#discussion_r1051193348) I was compiling it using the 3.0 version.

File: examples/dns/README.md (hunk 2 above)
Comment (Irene-123) [reply examples/dns/README.md:13]: This exact paragraph was written in the comments itself of `dns.sol` and I've grabbed it from the file. But I think we can make it more clear and show what are the parameters this contract uses.

File: examples/dns/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+![Image](dns_diagram.png)
+
+## Functionalities provided: 
+
+- Registration of a new Domain Name
+- Transfer the (already exsiting) owner to new Address
+- Change the Domain Name
+
+### Other functionalities to be implemented: 
+- Transfer bidding
```
[the remark this answers is not in the document]
Comment (Irene-123) [reply examples/dns/README.md:25]: Sure, I'll remove it then for now

File: examples/dns/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# DNS Smart contract
+
+The DNS smart contract is our showcase for executing other smart contracts on-chain.
+
+   
+
+> This is a port from the blog post's ink! 1.0 based version of the contract to ink! 2.0.
```
[the remark this answers is not in the document]
Comment (Irene-123) [reply examples/dns/README.md:7]: Okay, I've mentioned it in the readme. Also, I will try compiling it against 4.0 beta

File: examples/dns/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# DNS Smart contract
+
+The DNS smart contract is our showcase for executing other smart contracts on-chain.
```
[the remark this answers is not in the document]
Comment (Irene-123) [reply examples/dns/README.md:3]: Sure, changed it

**Comment by Irene-123** [comment]
Hii @HCastano I have made some changes, please review it and let me know! Thank you

**Comment by HCastano** [comment]
Hey @Irene-123 so taking another looks it seems like the info in the README is already mostly covered by the doc comments in the code itself. I'm gonna close it for that reason. Thanks for the effort though!

**Pull request closed without merging**